### DISCUSSION TITLE
Function Level Transaction

### DISCUSSION THREAD

**Discussion by SantoshKumarSubedi** [mikro-orm/mikro-orm#3575] [Q&A]
Is there anything like CLS transactional hook?
https://github.com/odavid/typeorm-transactional-cls-hooked
If not, If it's possible to implement something like the one linked above(TYPEORM) for mikro as well?

**Comment by B4nan** [comment]
> Is there anything like CLS transactional hook?

Not really

>  If it's possible to implement something

Yes, pr welcome

The real question is, why do you want this? With MikroORM your insert/update/delete queries are wrapped in a transaction automatically when you flush. But I can still see a value in something like that for select queries inside transaction, or when you want to use pessimistic locking.

FYI MikroORM already handles transactional async context propagation out of box (which is part of what that module does, via cls-hooked package).
